plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -405,3 +407,66 @@ def save(self):
 
     def __hash__(self):
         return hash(self.relative_path)
+
+
+class NopDownloader(BaseDownloader):
+    """
+    A no-operation (NOP) downloader.
+    """
+
+    def __init__(self):
+        super().__init__('')
+
+    def _send(self):
+        pass
+
+    async def run(self):
+        pass
+
+
+class NopPendingArtifact(PendingArtifact):
+    """
+    No operation (NOP) pending artifact.
+    """
+
+    def __init__(self, content):
+        """
+        Args:
+            content (PendingContent): The associated pending content.
+        """
+        super().__init__(Artifact(), '', '')
+        self.content = content
+
+    def downloader(self):
+        return NopDownloader()
+
+    def settle(self):
+        pass
+
+    def save(self):
+        pass
+
+
+class DownloadTask(asyncio.Task):
+    """
+    Specialized task (future) needed to support creating an artifact
+    after the file has been downloaded.
+    """
+
+    def __init__(self, artifact, downloader):
+        """
+        Args:
+            artifact (PendingArtifact):
+            downloader (BaseDownloader):
+        """
+        def done(task):
+            try:
+                task.exception()
+            except Exception:
+                pass
```

Comment:
Should this log here or is the point to silence exceptions and let the plugin writer report them at the end?

Reply:
I chose not to log the exception here thinking it would be the responsibility of the code using this iterator to handle the exception as seen fit.

Reply:
Sounds good.